libres/lib/analysis/analysis_module.cpp

```diff
@@ -73,11 +74,36 @@ analysis_module_type *analysis_module_alloc_named(int ens_size,
 
 analysis_module_type *analysis_module_alloc(int ens_size,
                                             analysis_mode_enum mode) {
-    if (mode == ENSEMBLE_SMOOTHER)
-        return analysis_module_alloc_named(ens_size, mode, "STD_ENKF");
-    else if (mode == ITERATED_ENSEMBLE_SMOOTHER)
-        return analysis_module_alloc_named(ens_size, mode, "IES_ENKF");
-    else
+    if (mode == ENSEMBLE_SMOOTHER) {
+        analysis_module_type *module = new analysis_module_type();
+        module->mode = mode;
+        module->user_name = util_alloc_string_copy("STD_ENKF");
+        module->module_config = std::make_unique<ies::config::Config>(false);
+        module->module_data = std::make_unique<ies::data::Data>(ens_size);
+        module->keys = {ies::data::ITER_KEY,
+                        ies::config::IES_INVERSION_KEY,
+                        ies::config::IES_LOGFILE_KEY,
+                        ies::config::IES_DEBUG_KEY,
+                        ies::config::IES_AAPROJECTION_KEY,
+                        ies::config::ENKF_TRUNCATION_KEY};
+        return module;
+    } else if (mode == ITERATED_ENSEMBLE_SMOOTHER) {
+        analysis_module_type *module = new analysis_module_type();
+        module->mode = mode;
+        module->user_name = util_alloc_string_copy("IES_ENKF");
+        module->module_config = std::make_unique<ies::config::Config>(true);
+        module->module_data = std::make_unique<ies::data::Data>(ens_size);
+        module->keys = {ies::data::ITER_KEY,
+                        ies::config::IES_MAX_STEPLENGTH_KEY,
+                        ies::config::IES_MIN_STEPLENGTH_KEY,
+                        ies::config::IES_DEC_STEPLENGTH_KEY,
+                        ies::config::IES_INVERSION_KEY,
+                        ies::config::IES_LOGFILE_KEY,
+                        ies::config::IES_DEBUG_KEY,
+                        ies::config::IES_AAPROJECTION_KEY,
+                        ies::config::ENKF_TRUNCATION_KEY};
+        return module;
+    } else
         throw std::logic_error("Undandled enum value");
```

Comment:
Not yours, but maybe fix typo when you are at it?

Reply:
Kinda looks like a real word...
Good catch, thanks.